.github/workflows/ci.yaml

```diff
@@ -39,7 +39,7 @@ jobs:
           ~/.cabal/store
           dist-newstyle
         key: |
-          cabal-${{ runner.os }}-${{ hashFiles('cabal.project') }}
+          cabal-${{ runner.os }}-${{ hashFiles('cabal.project', 'default.nix', 'shell.nix') }}
         restore-keys: |
           cabal-${{ runner.os }}
```

Comment:
Sorry about the PR ping pong @kk-hainq, but you would need to remove the `restore-keys` entry here. Otherwise the cache restores from any last built `cabal-ubuntu` cache key.

Reply:
Sorry, it was just a quick test run ~while I was in a meeting~... I was waiting to see the CI log before replying. It was a really great idea, I just wonder if we could improve by decoupling Nix and Cabal in this setup. Nix solves this dependency cache problem very well, but I guess it's still nice to test cabal commands in the nix shell in CI. Either way, we can always continue it in a separate issue later.